File: Serene/Serene.Script/Northwind/Shipper/ShipperDialog.cs

namespace Serene.Northwind
{
    using jQueryApi;
    using Serenity;
    using System.Collections.Generic;
    using System.Runtime.CompilerServices;

    [Imported]
    [IdProperty("ShipperID"), NameProperty("CompanyName")]
    [FormKey("Northwind.Shipper"), LocalTextPrefix("Northwind.Shipper"), Service("Northwind/Shipper")]
    public class ShipperDialog : EntityDialog<ShipperRow>
    {
    }
}

File: Serene/Serene.Web/Modules/Northwind/Shipper/ShipperDialog.ts
namespace Serene.Northwind {

    @Serenity.Decorators.registerClass()
    export class ShipperDialog extends Serenity.EntityDialog<ShipperRow, any> {
        protected getFormKey() { return ShipperForm.formKey; }
        protected getIdProperty() { return ShipperRow.idProperty; }
        protected getLocalTextPrefix() { return ShipperRow.localTextPrefix; }
        protected getNameProperty() { return ShipperRow.nameProperty; }
        protected getService() { return ShipperService.baseUrl; }

        protected form = new ShipperForm(this.idPrefix);

        protected getLanguages() {
            return LanguageList.getValue();
        }
    }
}